domain/user/service/service.go

```diff
@@ -19,46 +20,51 @@ import (
 type State interface {
```

Comment:
I have left a few comments around contracts on the state layer below. Can we make sure all the comments here are updated as well so their inline?